### PR TITLE
Ensure PGA works with pyganja

### PR DESCRIPTION
**Pull request by hugohadfield**

### CODE REVIEW

**Comment by eric-wieser** [comment]
So this doesn't actually add any new support, just improves the tests?

File: .gitignore [hunk 1]
```diff
@@ -102,3 +102,5 @@ venv.bak/
 
 # mypy
 .mypy_cache/
+
+\.idea/
```
Comment (eric-wieser) [inline .gitignore:106]: Why the backslash?

File: test_pyganja.py [hunk 2]
```diff
@@ -42,6 +42,49 @@ def test_draw_points(self):
         render_cef_script(str(gs), sig=layout.sig, grid=True, scale=1.0, gl=False)
 
 
+class TestPGA3DDrawing(unittest.TestCase):
+    def test_draw_points(self):
+        from clifford.pga import layout, blades
+        import numpy as np
+        def random_point():
+            return sum([c*b for c,b in zip(np.random.randn(3), [blades["e023"], blades["e013"], blades["e012"]])]) + blades["e123"]
```
Comment (eric-wieser) [inline test_pyganja.py:50]: nit:
```suggestion
            return sum(c*b for c,b in zip(np.random.randn(3), [blades["e023"], blades["e013"], blades["e012"]])) + blades["e123"]
```

File: test_pyganja.py [hunk 3]
```diff
@@ -42,6 +42,49 @@ def test_draw_points(self):
...
+        gs = GanjaScene()
+        gs.add_objects([random_point() for i in range(10)], static=False)
+        with open('test_file.html', 'w') as test_file:
+            print(generate_full_html(str(gs), sig=layout.sig, grid=True, scale=1.0, gl=True), file=test_file)
```
Comment (eric-wieser) [inline test_pyganja.py:54]: `print` adds extra newlines, you maybe want just:
```suggestion
            f.write(generate_full_html(str(gs), sig=layout.sig, grid=True, scale=1.0, gl=True))
```

File: test_pyganja.py [hunk 4]
```diff
@@ -42,6 +42,49 @@ def test_draw_points(self):
...
+        draw(gs, sig=layout.sig, grid=True, scale=1.0, gl=True)
+
+    def test_draw_lines(self):
+        from clifford.pga import layout, blades
+        import numpy as np
+
+        def random_point():
+            return sum([c * b for c, b in zip(np.random.randn(3), [blades["e023"], blades["e013"], blades["e012"]])]) + \
+                   blades["e123"]
```
Comment (eric-wieser) [inline test_pyganja.py:63]: May as well move this out, since you use it in all the tests.

File: .gitignore (hunk 1 above)
Comment (hugohadfield) [reply .gitignore:106]: not certain, it was autogenerated by gitkraken. Does it change the meaning?

**Comment by hugohadfield** [comment]
^ what is required here to make pga work is the update to the latest ganja.js version

**Comment by hugohadfield** [comment]
the rest is just testing and a comment in the readme

File: .gitignore (hunk 1 above)
Comment (eric-wieser) [reply .gitignore:106]: It doesn't mean anything, which is why none of the lines above have it
```suggestion
# PyCharm project settings
.idea/
```

File: test_pyganja.py [hunk 5]
```diff
@@ -38,10 +38,50 @@ def test_draw_points(self):
...
+        from clifford.pga import layout, blades
+        import numpy as np
+        self.layout = layout
+        self.blades = blades
+        self.np = np
+
+    def random_point(self):
+        return sum(c * b for c, b in
+                   zip(self.np.random.randn(3), [self.blades["e023"], self.blades["e013"], self.blades["e012"]])) + \
+               self.blades["e123"]
```
Comment (eric-wieser) [inline test_pyganja.py:57]: nit: could make it a bit shorter:
```suggestion
        return self.blades["e123"] + sum(
            c * self.blades[b] for c, b in zip(self.np.random.randn(3), ["e023", "e013", "e012"]
        )
```

**Comment by eric-wieser** [comment]
Trying to get github to recompute the diff

**Comment by eric-wieser** [comment]
Hmm. no dice. Can you rebase on master and force push to eliminate the merge commit within this patch?

**Comment by hugohadfield** [comment]
yep, one sec

**Pull request merged**